docs/advanced-features/using-mdx/next-mdx.md

```diff
@@ -0,0 +1,143 @@
+---
+description: Learn how to use @next/mdx in your Next.js project.
+---
+
+# `@next/mdx`
+
+The `@next/mdx` package is configured in the `next.config.js` file at your projects root. **It sources data from local files**, allowing you to create pages with a `.mdx` extension, directly in your `/pages` directory.
+
+### Setup `@next/mdx` in Next.js
+
+The following steps outline how to setup `@next/mdx` in your Next.js project:
+
+1. Install the required packages:
+
+   ```terminal
+     npm install @next/mdx @mdx-js/loader
+
+     #or
+
+     yarn add @next/mdx @mdx-js/loader
+   ```
+
+2. Require the package and configure to support top level `.mdx` pages. The following adds the `options` object key allowing you to pass in any plugins:
+
+   ```js
+   // next.config.js
+
+   const withMDX = require('@next/mdx')({
+     extension: /\.mdx?$/,
+     options: {
+       remarkPlugins: [],
+       rehypePlugins: [],
+     },
+   })
+   module.exports = withMDX({
+     pageExtensions: ['js', 'jsx', 'md', 'mdx'],
+   })
+   ```
+
+3. Create a new MDX page within the `/pages` directory:
+
+   ```bash
+     - /pages
+       - my-mdx-page.mdx
+     - package.json
+   ```
+
+## Using components, layouts and custom elements
+
+You can now import a React component directly inside your MDX page. Note that `@next/mdx` does **not** support frontmatter (frontmatter is a yaml like key, value pairing that can be used to store data about a page), instead, you can export data from within the `.mdx` file:
```

Comment:
this could maybe link to the mdx guide on how to support frontmatter with plugins: https://v2.mdxjs.com/guides/frontmatter/